File: backendApp/src/main/java/com/cinemas/dto/response/ItemIntroduce.java
package com.cinemas.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ItemIntroduce {
    private Integer id;
    private String name;
    private String slug;
    private String description;
    private String imagePortrait;
    private String trailer;

    public ItemIntroduce(String slug, String description) {
        this.slug = slug;
        this.description = description;
    }

    public ItemIntroduce(String slug, String description, String imagePortrait, String name) {
        this.slug = slug;
        this.description = description;
        this.imagePortrait = imagePortrait;
        this.name = name;
    }

    public ItemIntroduce(Integer id, String name, String slug, String imagePortrait) {
        this.id = id;
        this.name = name;
        this.slug = slug;
        this.imagePortrait = imagePortrait;
    }

//    public ItemIntroduce(Integer id, String name, String slug, String imagePortrait, String description) {
//        this.id = id;
//        this.name = name;
//        this.slug = slug;
//        this.imagePortrait = imagePortrait;
//        this.description = description;
//    }
    public ItemIntroduce(Integer id, String name, String slug, String imagePortrait, String trailer) {
        this.id = id;
        this.name = name;
        this.slug = slug;
        this.imagePortrait = imagePortrait;
        this.trailer = trailer;
    }
}

File: fllmbooking_app/lib/data/models/item_introduce.dart
class ItemIntroduce {
  final int id;
  final String name;
  final String slug;
  final String imagePortrait;
  final String trailer;

  ItemIntroduce({
    required this.id,
    required this.name,
    required this.slug,
    required this.imagePortrait,
    required this.trailer,
  });

  factory ItemIntroduce.fromJson(Map<String, dynamic> json) {
    return ItemIntroduce(
      id: json["id"] ?? 0,
      name: json["name"] ?? '',
      slug: json["slug"] ?? '',
      imagePortrait: json["imagePortrait"] ?? '',
      trailer: json["trailer"] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "slug": slug,
      "imagePortrait": imagePortrait,
      "trailer": trailer,
    };
  }

  @override
  String toString() {
    return 'ItemIntroduce{id: $id, name: $name, slug: $slug, imagePortrait: $imagePortrait, trailer: $trailer}';
  }
}
